internal/files/repository.go

```diff
@@ -5,11 +5,33 @@
 package files
 
 import (
+	"bytes"
 	"fmt"
 	"os"
 	"path/filepath"
+
+	"gopkg.in/yaml.v3"
 )
 
+func isGitWorktree(path string) error {
+	content, err := os.ReadFile(path)
+	if err != nil {
+		return fmt.Errorf("failed to read %s: %w", path, err)
+	}
+	type gitWorktree struct {
+		GitDir string `yaml:"gitdir"`
+	}
+	var worktree gitWorktree
+	dec := yaml.NewDecoder(bytes.NewBuffer(content))
+	dec.KnownFields(true)
+
+	if err := dec.Decode(&worktree); err != nil {
+		return fmt.Errorf("failed to decode %s: %w", path, err)
+	}
+
```

Comment:
Maybe we should also check if `worktree.GitDir` is not empty?

Reply:
Sure, I'll add this